src/interactions.js

```diff
@@ -57,7 +57,12 @@ module.exports = function(component, renderer, container) {
 	tmp_interactions.lastX = e.clientX;
 	tmp_interactions.lastY = e.clientY;
 
-	speckView.rotate(component.props.view, dx, dy);
+	var view = Object.assign({}, component.props.view);
```

Comment:
We actually try to avoid using `var` in favour of using `let` and `const`: https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75

Reply:
I've also run into errors when some parameters of `view` aren't defined, so it might be better to do something like

```suggestion
	var view = Object.assign(speckView.new(), component.props.view);
```

in case the user supplies only a subset of parameters (e.g., zoom and resolution, like I had in the app).

Reply:
Ya I try to use `let` and `const` instead of `var` too in general where possible, however I noted that speck is written in ES5 JS and does not contain ES6 JS (a newer version of JS) which is the version where `let` and `const` was introduced. I thought it might be better to stick to ES5 as result, although let me know if you would still like me to change to `let` and `const`.

Reply:
Interesting! Do you anticipate using `let/const` becoming an issue in the future, or this change just to remain consistent with the style of the other code in Speck? If it's the latter, I personally don't think we should change to `var`, since `let` and `const` are (IMO) easier to read. If it's the former, then we should definitely make that change :)

Reply:
No there is nothing inherently bad about using `var` vs `let/const`, it is just easier to make coding mistakes when using `var` which is why I and I suspect most people prefer to use `let/const` when given the choice. I personally don't see a reason to make the change to the PR (which uses `var`) since the rest of the repo uses ES5 already and introducing ES6 for just 1 variable might be a consistency problem :)